### ISSUE TITLE
GetPathTemplate() crashes on Solaris

### ISSUE THREAD

**Issue by jeroen** [ImageMagick/ImageMagick#1503]
This is not very important but nevertheless.... I use the following function to set the temporary path. It seems OK on all platforms except for Solaris where it crashes at `GetPathTemplate()`. What could be the reason? 

```c
char * set_magick_tempdir(const char * new_tmpdir){
    MagickCore::ExceptionInfo *exception = MagickCore::AcquireExceptionInfo();
    MagickCore::SetImageRegistry(MagickCore::StringRegistryType, "temporary-path", new_tmpdir, exception);
    exception=DestroyExceptionInfo(exception);
  
  //Try to read current tempdir
  static char path[4000] = "";
#if MagickLibVersion >= 0x681
  MagickCore::GetPathTemplate(path);
#endif
  return path;
}
```

**Comment by jeroen** [comment]
Hmmm actually it seems to be segfaulting for anything with paths...

**Comment by dlemstra** [comment]
The size of your buffer is not big enough? Should be MaxTextExtent (4096).

**Comment by jeroen** [comment]
Yeah I don't think this was the issue, IM7 actually crashes on Solaris for almost every operation now.

**Comment by urban-warrior** [comment]
Which operations?  We ran the unit tests on Solaris 11-4 with gcc 4.7 and only 1 failed, the other thousands of unit tests passed.

**Comment by jeroen** [comment]
Anything using the IM7 C++ api crashes on Solaris 10 for me. A simple program like this:

```cpp
#include <Magick++.h>
int main(){
  Magick::Image image("logo:");
  image.border("10x10");
  image.write(std::string("test.png"));
}
```

Core dumps immediately.
<img width="1026" alt="Screen Shot 2019-03-14 at 10 10 39 AM" src="https://user-images.githubusercontent.com/216319/54344575-75854e80-4641-11e9-8783-ecb36fc162b0.png">

**Comment by dlemstra** [comment]
Can you get more info with gdb?

**Comment by jeroen** [comment]
It's Solaris so my tools are a bit limited to say the least :-) Does this help:

<img width="930" alt="Screen Shot 2019-03-14 at 10 17 40 AM" src="https://user-images.githubusercontent.com/216319/54345013-710d6580-4642-11e9-9ee5-fbcf78d4b5aa.png">

**Comment by urban-warrior** [comment]
Unfortunately, we cannot reproduce the problem.  We used Solaris 10.  We ensured we were linking against the installed version of ImageMagick instead of the one installed in /usr/sfw/bin.  Your example program ran without throwing an exception and returned the test.png output file as expected.

**Comment by jeroen** [comment]
Can you try the version of imagemagick from opencsw?

```
/opt/csw/bin/pkgutil -i imagemagick_dev
```

**Comment by urban-warrior** [comment]
We would rather not.  Its a build we have no control over, it may not have been built correctly.  We installed ImageMagick from scratch to prove it is working properly.  If other builds fail, its either a problem with the build or a problem with those linking to the build-- otherwise operator error, something we have no control over.

**Comment by jeroen** [comment]
Sure I see that. 

The problem appears both in the opencsw version of IM7 as well as the one built manually installed on the CRAN check servers. It does not appear with IM6. So there does seem some regression here but perhaps only for a certain config...

**Comment by urban-warrior** [comment]
We've proven, ImageMagick 7 works under Solaris 11 and Solaris 10.  We're clueless why it fails for the CSW build and in particular why IM6 works but IM7 fails.  Which release does CSW use?  Can we get a build with the latest IM release IM 7.0.8-34?  It has all the latest patches, perhaps it will work.

**Comment by jeroen** [comment]
It uses 7.0.8-29 right now. But the issue has been here for quite a while. I'll try to ask the OpenCSW maintainer to help getting a debug build, perhaps that gives us a clue what is going wrong.

**Comment by urban-warrior** [comment]
Appreciate that.  We develop under Linux/Windows.  Its a pain to stand up Solaris, we had to use a VM template to install and its quite time consuming-- thus the reluctance to go deeper given we've proven ImageMagick is working under Solaris.
